xdsl/transforms/experimental/stencil_tensorize_z_dimension.py

```diff
@@ -261,7 +267,9 @@ def is_tensorized(
     return len(typ.get_shape()) == 2 and isinstance(typ.get_element_type(), TensorType)
 
 
-def is_tensor(typ: Attribute) -> TypeGuard[TensorType[Attribute]]:
+def is_tensor(
+    typ: Attribute,
+) -> TypeGuard[TensorType[IndexType | IntegerType | AnyFloat]]:
```

Comment:
why this change?

Reply:
sorry if this is incorrect, as i'm not fully understanding what this pass does, but it looked like `is_tensor` check is used whether an operand is a tensor and can be rewritten as `DenseIntOrFPElementsAttr`, but `Tensortype[Attribute]` is not good enough of a constraint for that, as the dense type can only store tensors of these values.

I'm not sure why pyright didn't complain about this before, I'll try to investigate

Reply:
Oh it didn't complain before because there is no check when you don't use the custom constructors

Reply:
Ah nice, I see